MonoGamePico8/backend/MonoGameGraphicsBackend.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.IO;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Pico8Emulator;
+using Pico8Emulator.backend;
+using Pico8Emulator.unit.graphics;
+using Pico8Emulator.unit.mem;
+
+namespace MonoGamePico8.backend {
+	public class MonoGameGraphicsBackend : GraphicsBackend {
+		public Texture2D Surface;
+		
+		private GraphicsDevice graphics;
+		private Color[] screenColorData = new Color[GraphicsUnit.ScreenSize];
+		private Color[] palette;
+
+		public MonoGameGraphicsBackend(GraphicsDevice graphicsDevice) {
+			graphics = graphicsDevice;
+			palette = new Color[Palette.Size];
+
+			for (var i = 0; i < Palette.Size; i++) {
+				palette[i] = new Color(Palette.StandardPalette[i, 0], Palette.StandardPalette[i, 1], Palette.StandardPalette[i, 2]);
+			}
+		}
+		
+		public override void CreateSurface() {
+			Surface = new Texture2D(graphics, 128, 128, false, SurfaceFormat.Color);
+		}
+
+		public override void Flip() {
+			var ram = Emulator.Memory.Ram;
+			var drawState = Emulator.Memory.DrawState;
+
+			for (var i = 0; i < 8192; i++) {
+				var val = ram[i + RamAddress.Screen];
+
+				screenColorData[i * 2] = palette[drawState.GetScreenColor(val & 0x0f)];
+				screenColorData[i * 2 + 1] = palette[drawState.GetScreenColor(val >> 4)];
+			}
+			
+			Surface.SetData(screenColorData);
+		}
+		
+		private byte ColorToPalette(Color col) {
+			for (var i = 0; i < 16; i += 1) {
+				if (palette[i] == col) {
+					return (byte) i;
+				}
+			}
+
+			return 0;
+		}
+
+		public override void Import(string path, bool onlyHalf) {
+			var texture = Texture2D.FromStream(graphics, new FileStream(path, FileMode.Open));
+			
+			if (texture.Height != 128 || texture.Width != 128) {
+				throw new ArgumentException($"{path} must be a 128x128 image, but is {texture.Width}x{texture.Width}.");
+			}
+
+			var bound = onlyHalf ? 64 : 128;
+			var data = new Color[GraphicsUnit.ScreenSize];
+
+			texture.GetData(data);
+
+			for (var i = 0; i < bound; i += 1) {
+				for (var j = 0; j < 128; j += 1) {
+					Emulator.Graphics.Sset(j, i, ColorToPalette(data[j + i * 128]));
+				}
+			}
+
+			texture.Dispose();
+		}
+
+		public override void Export(string path) {
+			var texture = new Texture2D(graphics, 128, 128, false, SurfaceFormat.Color);
+			var data = new Color[GraphicsUnit.ScreenSize];
+			
+			for (var i = 0; i < 128; i += 1) {
+				for (var j = 0; j < 128; j += 1) {
+					data[j + i * 128] = palette[Emulator.Graphics.Sget(j, i)];
+				}
+			}
+
+			texture.SetData(data);
+			texture.SaveAsPng(File.Create(path), 128, 128);
+			texture.Dispose();
+		}
```

Comment:
I think we can simplify the graphics backend and not include import/export. The old implementation was already monogame free since it used Color from System.Drawing! WDYT ? Was there any reason for this to be done this way?

Reply:
It still requires palette... And palette is only accessible here.

Reply:
In this case we can have another separate palette that is defined using System.Drawing.Color and use that instead. The old implementation handles all of that already ([here](https://github.com/mmortatti/pico8-emulator/blob/84833683849ea77a0cde18b954b9488f65154687/Pico8.cs)). It doesn't define a whole palette, it creates color values on the fly using System.Drawing.Color.FromArgb, so that can be improved. However I don't think people will be calling Export all the time, so if there are multiple performance changes to those two functions we can create an issue and leave that for later.